test/attempts.test.js

```diff
@@ -23,4 +24,19 @@ describe('Request attempts', function () {
       done();
     });
   });
+
+  it('should call delay strategy 2 times after some retries', function (done) {
+    var mockDelayStrategy = sinon.stub().returns(100);
+
+    request({
+      url: 'http://www.filltext.com/?rows=1&err=500', // return a 500 status
+      maxAttempts: 3,
+      retryDelay: 100,
```

Comment:
Ah ok. Sorry I misunderstood you. Thx for breaking it down for me. I have checked in the changes. Please have a look and let me know if anything needs to be changed. Appreciate it = )